src/DataStructures/Tensor/Expressions/TimeIndex.hpp

```diff
@@ -0,0 +1,96 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <type_traits>
+
+#include "DataStructures/Tensor/Expressions/TensorIndex.hpp"
+#include "DataStructures/Tensor/IndexType.hpp"
+#include "DataStructures/Tensor/Symmetry.hpp"
+#include "Utilities/Algorithm.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \file
+/// Defines functions and metafunctions used for helping evaluate
+/// TensorExpression equations where generic spatial indices are used for
+/// spacetime indices
+
+namespace tt {
+/// \ingroup TypeTraitsGroup TensorExpressionsGroup
+/// \brief Check if a type `T` is a TensorIndex representing a concrete time
+/// index
+template <typename T>
+struct is_time_index : std::false_type {};
+template <>
+struct is_time_index<std::decay_t<decltype(ti_t)>> : std::true_type {};
+template <>
+struct is_time_index<std::decay_t<decltype(ti_T)>> : std::true_type {};
+}  // namespace tt
+
+namespace TensorExpressions {
+namespace detail {
+/// \brief Returns whether or not the provided value is the TensorIndex value
+/// that encodes the upper or lower concrete time index (`ti_T` or `ti_t`)
+///
+/// \param value the value to check
+/// \return whether or not the value encodes the upper or lower concrete time
+/// index
+constexpr bool is_time_index_value(const size_t& value) {
```

Comment:
fundamentals like `size_t`, `double` you can just take by value instead of reference. A reference is a pointer (8 bytes) and a size_t is 8 bytes, so there's not anything being saved by a pointer. I'd guess some compiler optimization passes even eliminate references in this case